### ISSUE TITLE
Different results of halfcheetah

### ISSUE THREAD

**Issue by rainbow979** [anuragajay/decision-diffuser#1]
Hello,

I tried the same config with the repo and got the same good performance with the paper. However, when I tried the env halfcheetah and the testing score is much lower than the results from the paper. I only changed the returns_scale from 400 to 800 since halfcheetah had higher discounted returns. The training loss is shown in the figure. The blue line is halfcheetah and the purple line is hopper. I am wondering if there are other hyperparemeters needs to be changed.
![1673933898018](https://user-images.githubusercontent.com/16297710/212818478-e54bd25a-f554-4c0b-a6c4-0656be13c889.png)

**Comment by WangYong-Design** [comment]
Have you kept track of how long it takes to run an experiment on GPU?

**Comment by Looomo** [comment]
Hi, I was trying to reproduce the results of DD, but I couldnt make it, results on each dataset differ significantly from the results presented in the paper in Tab1. （I have tried only on 4 datasets, walker2d-medium-replay-v2, walker2d-medium-v2,halfcheetah-medium-replay-v2, and halfcheetah-medium-v2）. So have you change anything in this repo or just directly run the downloaded code? Thanks!
Also, did you find the codes of  length-K history condition in this repo?

**Comment by RenMing-Huang** [comment]
same question

**Comment by SpaceLearner** [comment]
same question

**Comment by SpaceLearner** [comment]
same question

**Comment by xishuxishu** [comment]
> Hello,
> 
> I tried the same config with the repo and got the same good performance with the paper. However, when I tried the env halfcheetah and the testing score is much lower than the results from the paper. I only changed the returns_scale from 400 to 800 since halfcheetah had higher discounted returns. The training loss is shown in the figure. The blue line is halfcheetah and the purple line is hopper. I am wondering if there are other hyperparemeters needs to be changed. ![1673933898018](https://user-images.githubusercontent.com/16297710/212818478-e54bd25a-f554-4c0b-a6c4-0656be13c889.png)

hello，Have you solved this problem

**Comment by wangerlie** [comment]
> Hello,
> 
> I tried the same config with the repo and got the same good performance with the paper. However, when I tried the env halfcheetah and the testing score is much lower than the results from the paper. I only changed the returns_scale from 400 to 800 since halfcheetah had higher discounted returns. The training loss is shown in the figure. The blue line is halfcheetah and the purple line is hopper. I am wondering if there are other hyperparemeters needs to be changed. ![1673933898018](https://user-images.githubusercontent.com/16297710/212818478-e54bd25a-f554-4c0b-a6c4-0656be13c889.png)

hello @rainbow979 
I also tried to reproduce the results, I run the code directly without ANY change in the code, I just couldn't get the same result in 'hopper-medium-expert-v2'. I am wondering how to reproduce the result in 'hopper-medium-expert-v2', have you ever change any hyperparameters? Thanks a lot .

**Comment by wangerlie** [comment]
I am also wondering do you have the same warning as following, I think it's not reasonable, there may have some problem within the normalization code?
![image](https://github.com/user-attachments/assets/d8fe5bfc-a859-42bf-8db7-d1c05aefd98e)
